compiler/rustc_session/src/lib.rs

```diff
@@ -4,6 +4,7 @@
 #![feature(let_chains)]
 #![feature(map_many_mut)]
 #![feature(rustc_attrs)]
+#![recursion_limit = "256"]
```

Comment:
Suggestion: could you please leave a comment for why this recusion limit is needed, as you explained in https://github.com/rust-lang/rust/pull/133138#issuecomment-2551349559?

Reply:
Fixed.